pkg/engine/shield/add.go

```diff
@@ -3,21 +3,65 @@ package shield
 import (
 	"github.com/simimpact/srsim/pkg/engine/event"
 	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/prop"
 	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
 )
 
 func (mgr *Manager) AddShield(id key.Shield, shield info.Shield) {
-	// TODO: LOGIC FOR ADDING A SHIELD
-	// 1. Check if the target already has this shield, if so remove old (mgr.RemoveShield)
-	// 2. Get the stats for the source and target
-	// 3. Compute shield HP/create ShieldInstance given the add paramsm
-	// 4. add shield to mgr.targets[shield.target]
-	// 5. emit ShieldAdded event
-
-	// emit to signify shield added
+
+	// Get the stats for the source and target
+	source := mgr.attr.Stats(shield.Source)
+	maxShield := mgr.targets[shield.Target][mgr.MaxShield(shield.Target)]
```

Comment:
this will cause the sim to crash if this target has no shields (should be 0 if they have no shields)